File: tests/go/cases/case_265_interface_sink_bridge_06.go
package escape_tests

var retainedCase265 = []map[string]string{}

type case265Sinker interface{ Put(map[string]string) }
type case265Bridge struct{}

func (case265Bridge) Put(v map[string]string) {
	retainedCase265 = append(retainedCase265, v)
}

func Case265InterfaceSinkBridge06(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "interface_sink_bridge_06",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "interface_sink_bridge_06:" + raw,
	}
	var s case265Sinker = case265Bridge{}
	// ESCAPE: interface bridge dispatch to sink.
	s.Put(payload)
	return "ok"
}

File: tests/nodejs/cases/case_265_interface_sink_bridge_06.js
/**
 * Task case 265: interface_sink_bridge_06 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case265InterfaceSinkBridge06(input) {
  const taskName = 'interface_sink_bridge_06';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const sink = { put(obj) { retainedAudit.push(obj); } };
  // ESCAPE: object method bridge to retained sink.
  sink.put(payload);
  return 'ok';
}

module.exports = { case265InterfaceSinkBridge06 };
